cita-executor/core/src/cita_executive.rs

```diff
@@ -256,50 +259,165 @@ impl<B: DB + 'static> CitaExecutive<B> {
+        };
+        Ok(receipt)
+    }
+}
+
+fn build_evm_transaction(params: &VmExecParams) -> EVMTransaction {
+    EVMTransaction {
+        from: params.sender,
+        value: params.value,
+        gas_limit: params.gas.as_u64(),
+        gas_price: params.gas_price,
+        input: params.data.clone(),
+        to: params.to_address,
+        nonce: params.nonce,
+    }
+}
+
+fn build_evm_context(env_info: &EnvInfo) -> EVMContext {
+    EVMContext {
+        gas_limit: env_info.gas_limit.as_u64(),
+        coinbase: env_info.author,
+        number: U256::from(env_info.number),
+        timestamp: env_info.timestamp,
+        difficulty: env_info.difficulty,
+    }
+}
+
+fn build_evm_config(block_gas_limit: u64) -> VMConfig {
+    VMConfig {
+        // block_gas_limit is meaningless in cita_vm, so let it as default_block_quota_limit.
+        block_gas_limit,
+        ..Default::default()
+    }
+}
+
+fn build_receipt_with_ok(init_gas: U256, result: InterpreterResult) -> Receipt {
+    let mut receipt = Receipt::default();
+
+    match result {
+        InterpreterResult::Normal(_data, quota_left, logs) => {
+            receipt.quota_used = init_gas.as_u64() - quota_left;
+            receipt.logs = transform_logs(logs);
+            receipt.logs_bloom = logs_to_bloom(&receipt.logs);
+        }
+        InterpreterResult::Revert(_data, quota_left) => {
+            receipt.quota_used = init_gas.as_u64() - quota_left;
+        }
+        InterpreterResult::Create(_data, quota_left, logs, contract_address) => {
+            receipt.quota_used = init_gas.as_u64() - quota_left;
+            receipt.logs = transform_logs(logs);
+            receipt.logs_bloom = logs_to_bloom(&receipt.logs);
+
+            let address_slice: &[u8] = contract_address.as_ref();
+            receipt.contract_address = Some(H160::from(address_slice));
+        }
+    };
+    receipt
+}
+
+fn build_receipt_with_err(err: VMError) -> Receipt {
+    let mut receipt = Receipt::default();
+    receipt.receipt_error = err.to_string();
+    receipt
+}
+
+fn transform_logs(logs: Vec<EVMLog>) -> Vec<LogEntry> {
+    logs.into_iter()
+        .map(|log| {
+            let EVMLog(address, topics, data) = log;
+
+            LogEntry {
+                address,
+                topics: topics
+                    .into_iter()
+                    .map(|topic| Hash::from_bytes(topic.as_ref()).expect("never returns an error"))
+                    .collect(),
+                data,
+            }
+        })
+        .collect()
+}
+
+fn logs_to_bloom(logs: &[LogEntry]) -> Bloom {
+    let mut bloom = Bloom::default();
+
+    logs.iter().for_each(|log| accrue_log(&mut bloom, log));
+    bloom
+}
+
+fn accrue_log(bloom: &mut Bloom, log: &LogEntry) {
+    // FIXME: upgrade ethereum-types to 0.6.0 to avoid unsafe block.
+    unsafe {
+        bloom.accrue(BloomInput::Raw(slice::from_raw_parts(
+            log.address.as_ptr(),
+            20,
+        )));
+    }
+
```

Comment:
```rs
bloom.accrue(BloomInput::Raw(log.address.as_bytes()));
```
In m*, There must be a safe type converstion  from `address-> bytes`, maybe `copy_to` or somewhat.

Reply:
Yes, have you seen my FIXME description?
```
// FIXME: upgrade ethereum-types to 0.6.0 to avoid unsafe block.
```